### PR TITLE
STR-440 OpenTelemetry tracing exporter

### PR DESCRIPTION
**Pull request by delbonis**
## Description

This PR is to support the OpenTelemetry exporter so we can have richer logging from our production environment.

### Type of Change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature/Enhancement (non-breaking change which adds functionality or enhances an existing one)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation update
-   [ ] Refactor

## Checklist

-   [ ] I have performed a self-review of my code.
-   [ ] I have commented my code where necessary.
-   [ ] I have updated the documentation if needed.
-   [ ] My changes do not introduce new warnings.
-   [ ] I have added tests that prove my changes are effective or that my feature works.
-   [ ] New and existing tests pass with my changes.

### CODE REVIEW

**Comment by delbonis** [comment]
Alright so it builds and inits properly, but now we're getting this issue when we try to run the client in a local test env with a url to a local Jaeger instance running:

```
OpenTelemetry trace error occurred. Exporter otlp encountered the following error(s): the grpc server returns error (Unknown error): , detailed error message: transport error tonic::transport::Error(Transport, hyper::Error(Io, Kind(ConnectionReset)))
```

We can run Jaeger now with `./contrib/jaeger.sh`, so to replicate this issue just run that and run the functional tests normally with `./run_tests.sh fn_sync_genesis`.

File: Cargo.toml [hunk 1]
```diff
@@ -206,10 +208,11 @@ tempfile = "3.10.1"
 terrors = "0.3.0"
 thiserror = "1.0"
 threadpool = "1.8"
-tokio = { version = "1.37", features = ["full"] }
+tokio = { version = "1.*", features = ["full"] }
```
Comment (storopoli) [inline Cargo.toml:211]: We should pin tokio.
This is fine for scaffolding the PR.
Imagine two versions of tokio, the huge compile time...
I know that we're somewhat between a rock and a hard place with https://github.com/tokio-rs/tracing-opentelemetry/issues/170

Comment (delbonis) [reply Cargo.toml:211]: Oh I think I forgot to fix this.  I changed it while I was trying to iron out the dependency mixups.

**Comment by delbonis** [comment]
Alright I tried it with [opentelemetry-collector](https://opentelemetry.io/docs/collector/quick-start/) and it seems like Jaeger was just not listening for some inexplicable reason.  This doesn't give the nice UI but it does show that the trace events are being generating by spamming strata-looking strings out in the docker output.  I will investigate further but it's good to know it seems like this is merely a configuration issue now.

**Comment by delbonis** [comment]
Figured out the Jaeger thing, I was just using an ancient version of Jaeger because I copypasted a docker command from their docs that wasn't recent, so it didn't support the OpenTelemetry collector.

**Comment by delbonis** [comment]
Should figure out a better solution to the logging in tests, maybe should make a simpler stub.

File: bin/strata-client/src/extractor.rs [hunk 2]
```diff
@@ -218,7 +218,8 @@ mod tests {
 
     #[tokio::test]
     async fn test_extract_deposit_requests() {
-        logging::init();
+        // FIXME this is absurd, why are we doing this here?
+        logging::init(logging::LoggerConfig::with_base_name("strata-client-tests"));
```
Comment (storopoli) [inline bin/strata-client/src/extractor.rs:222]: It this calling `btcio::client` by any means?
If yes there's a lot of good logging happening there that is quite useful.

**Pull request merged**